### PR TITLE
+ Addition of (final) 3D thermal particle model from the PhD work of …

### PR DESCRIPTION
**Pull request by JonMcCullough**
…J. McCullough

I added:
- the model: _d3q27_tePSM_per_ a total energy thermal model with particle interactions.
- periodicity in the y and z directions when coupling with E-SyS Particle

### CODE REVIEW

**Comment by JonMcCullough** [comment]
Hi Lukasz, 
Do you happen to have any idea why this is failing the compile checks?

We are looking to get the particles branch into the main repository and up to date with these changes. We have some new PhD students here that we want to get going on some benchmark testing.

Cheers,
Jon

**Comment by llaniewski** [comment]
Hi. Good to know. I was recently under a pile of soon-to-be articles to review.

The thing is that the tests on Travis-CI much more restrictive in compilation then normal compilation, so sometimes it catches errors that are not visible while you develop.

The error was found in the RemoteForceInterface. But in CFD-GO/TCLB#189 it went through so I probably corrected something.

I wanted to separate the branches as is CFD-GO/TCLB#192 before I'll introduce the support for particles in the main code. But that is done, so I think this week I will proceed with this, and then we'll merge your model.

**Comment by llaniewski** [comment]
@JonMcCullough No. I was mistaken. The bug is in the changes you introduced. I have to check it in detail.

Do you have a good testcase to check your periodicity changes?

File: src/MPMD.hpp [hunk 1]
```diff
@@ -3,6 +3,8 @@
 #include <string>
 #include <mpi.h>
 #include <assert.h>
+#include <map> //JM
+#include <string.h> //JM
```
Comment (llaniewski) [inline src/MPMD.hpp:7]: Why `<string.h>` when you have `<string>` ?
I have to migrate the changes to https://github.com/CFD-GO/MPI_MPMD

**Comment by JonMcCullough** [comment]
Thanks Lukasz. From memory, those two includes were required for me to compile things on Magnus. It could be ultimately unnecessary.

**Comment by llaniewski** [comment]
@JonMcCullough Tomorrow I'll be sitting on merging CFD-GO/TCLB#189 and this (#1), so I have two main questions:
- Do you have a good testcase to check your periodicity changes?
- What version of ESYS-Particles I should use for testing?

**Comment by JonMcCullough** [comment]
Unfortunately I do not have a good testcase to hand (I have just come to London - things are a bit disordered at the moment). My version of esys is the most up-to-date with changes for periodicity. I did post a merge request of it to you on launchpad. See https://code.launchpad.net/~jonmcc for details.

**Comment by llaniewski** [comment]
First step completed:
- merged `lp:~llaniewski/esys-particle/remote-force` with `lp:~jonmcc/esys-particle/remote-force-periodicity`
- merged `lp:~llaniewski/esys-particle/remote-force` with upstream repo `lp:esys-particle`

Current version with changes from @JonMcCullough and ESYS-Particle team in on https://code.launchpad.net/~llaniewski/esys-particle/remote-force

It's working with the current `particles` branch.

**Pull request merged**